tincr/io/device/family_info.tcl

```diff
@@ -0,0 +1,773 @@
+        if {[llength $toks] != 3} {
+           puts "\nParse Error: Invalid file format for " $vsrt_bels_file
+           exit
+        }
+        
+        set family_name [lindex $toks 0]
+        set site_name [lindex $toks 1]
+        set bel_name [lindex $toks 2]
+        
+        # We only expect VCC and GND BELs in this file, 
+        # if this is not the case, then throw an exception
+        if {[string first "GND" $bel_name] != -1} {
+            set type "GND"
+        } elseif {[string first "VCC" $bel_name] != -1} {
+            set type "VCC"
+        } else {
+            puts "[Parse Error]: The BELs in \"addedBels.txt\" should only be GND or VCC BELs. "
+            puts "\t $bel_name does not match this pattern."
+            exit
+        }
+        
+        # only include added bels for the current family
+        if {$family_name == $family} {
+            dict lappend vsrt_bels_map [string toupper $site_name] ${bel_name}:$type
+        }
+    }
+    
+    return $vsrt_bels_map
+}
+
+## Returns true if the specified site is a valid alternate.
+#   The type site should be set before this function is called.
+#
+# @param site Site handle
+proc is_valid_alternate_type { site } {
+    global is_series7
+    global invalid_alternate_types
+    if {$is_series7} {
+        set type [get_property SITE_TYPE $site]
+        if {[lsearch $invalid_alternate_types $type] != -1} {
+            return 0
+        }
+    }
+    return 1
+}
+
+## Creates a new familyInfo.xml file for the specified family
+#   This XML file contains additional device information/corrections
+#   that are not included in the XDLRC generated from <tincr::write_xdlrc>"()".
+#
+# @param filename Output XML file. If the file does not end in ".xml", then ".xml" will be appended.
+# @param family The family to generate the family info for. Possible options include:
+#  <p><ul>
+#  <li> artix7 
+#  <li> kintex7 
+#  <li> virtex7 
+#  <li> zynq 
+#  <li> kintexu 
+#  <li> kintexuplus
+#  <li> virtexu 
+#  <li> virtexuplus
+#  <li> zynquplus
+# </ul><p>
+# @param vsrt_bels_file (Optional) The VSRT "addedBels.txt" file generated
+#   while creating primitive defs in VSRT. If you don't know what this means, you can safely ignore it.
+proc ::tincr::create_xml_family_info { filename family {vsrt_bels_file ""} } {
+
+    catch { close_project -quiet  }
+    
+    # create the XML file
+    set filename [::tincr::add_extension ".xml" $filename]    
+    set fileout [open $filename w]
+    
+    # parse the addedBelsDirectory
+    set vsrt_bel_map [parse_vsrt_bels $family $vsrt_bels_file]
+    
+    # print header
+    print_header_family_info $family $fileout
+    
+    # Initialize variables
+    set primary_set [list]
+    set alternate_set [list]
+    set unique_parts [tincr::get_parts_unique $family]
+    set i 1
+    global is_series7
+    set is_series7 [expr {[string first "7" $family] != -1}] 
```

Comment:
This operation is cheap. I would recommend putting it into an ensemble (e.g. tincr/cad/device/devices.tcl) and just call it from there each time a proc needs to know if the current device is Series7.